zenoh/src/net/routing/dispatcher/queries.rs

```diff
@@ -718,7 +718,7 @@ pub(crate) fn route_send_response(
                         ext_tstamp,
                         ext_respid,
                     },
-                    "".to_string(), // @TODO provide the proper key expression of the response for interceptors
+                    key_expr.to_string(),
```

Comment:
This allocates everytime in the critical path. Why it should return a `String` and not a `keyexpr`?

Reply:
It's because `RoutingContext<Msg>` expects a `String`. In other instances where this is called no additional allocation is made, but here it is the case...
For more detail see here: https://github.com/eclipse-zenoh/zenoh/blob/main/zenoh/src/net/routing/mod.rs

Reply:
Let's then return empty `String` for the time being until a proper solution is found. 
Does returning an empty `String` impact the behaviour?

Reply:
Returning an empty string breaks the keyexpr matching (empty string does not match with any key expression, even `**`).
Also, rules are made on key expressions, so having an empty string on all replies would impact the behavior of ACL

Reply:
Ok, understood. Can you then verify the impact on performance?

Reply:
I'll look into it 👍

Reply:
Due to high impact on performance, this change will be reverted. We can use `Response{ wire_expr, ... }` in ACL code to only get the keyexpr when needed.